packages/superset-ui-core/src/chart/components/SuperChart.tsx

```diff
@@ -132,8 +135,33 @@ export default class SuperChart extends React.PureComponent<Props, {}> {
       enableNoResults &&
       (!queriesData ||
         queriesData.every(({ data }) => !data || (Array.isArray(data) && data.length === 0)));
+    let hasNullMetrics = false;
+    if (chartProps.formData?.disallowNullMetrics) {
```

Comment:
This name is a bit ambiguous in the sense that it's unclear what disallowing exactly means. This is a bit of a monster, but perhaps something along these lines could work? `showNoResultsOnAllNullMetrics` (ok that's really horrible, but you get the idea)